collector/process.go

```diff
@@ -191,7 +191,10 @@ func (c *ProcessCollector) collect(ch chan<- prometheus.Metric) (*prometheus.Des
 
 	var dst_wp []WorkerProcess
 	q_wp := queryAll(&dst_wp)
-	wmi.QueryNamespace(q_wp, &dst_wp, "root\\WebAdministration")
+	err := wmi.QueryNamespace(q_wp, &dst_wp, "root\\WebAdministration")
+	if err != nil {
+		log.Debugf("Could not query WebAdministration namespace for IIS worker processes: %v. Skipping", err)
```

Comment:
I don't like this, really. We should somehow be able to decide if this is good or bad and appropriately ignore or error on it. At the moment we'll always try this, even for machines without IIS, so we can't say that it is unexpected that it fails. On machines that do have IIS, however, this indicates a problem.

Should probably be fixed in another PR though.

Reply:
I agree this should be handled more fully, but as you say lets leave it for another PR